src/alpaca_parser.yrl

```diff
@@ -145,12 +150,36 @@ record_type_members -> record_type_member ',' record_type_members : ['$1' | '$3'
 record_type -> open_brace record_type_members close_brace : 
   #t_record{members='$2'}.
 
+module_qualified_type_name -> symbol '.' symbol:
+  {symbol, L, Mod} = '$1',
+  {symbol, _, Name} = '$3',
+  {module_qualified_type_name, L, Mod, Name}.
+
+module_qualified_type -> module_qualified_type_name :
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name}}.
+
+module_qualified_type -> module_qualified_type_name type_expressions:
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name},
+     vars = [V || {type_var, _, _}=V <- '$2']}.
```

Comment:
You're not missing anything, you're absolutely correct that we're dropping that information but it belongs in a type variable rather than as a member I think.  E.g.

    type opt 'a = Some 'a | None

Here `'a` is not actually a member of the type, it's the argument to `Some` which along with `None` are the members.  Whereas in the following:

    type something_else 'a = 'a | What 'a

`'a` is indeed a member.  I filed issue #113 for this but I'm entirely willing to believe it's not describing this sufficiently :)  Thoughts?